vortex-array/src/array/primitive/mod.rs

```diff
@@ -134,6 +141,65 @@ impl<'a> PrimitiveArray<'a> {
     pub fn into_buffer(self) -> Buffer<'a> {
         self.into_array().into_buffer().unwrap()
     }
+
+    fn subtract_scalar_integer<
+        T: NativePType + Display + OverflowingSub + Sub + CheckedSub + Clone,
+    >(
+        &self,
+        to_subtract: &Scalar,
+        should_wrap: bool,
+    ) -> VortexResult<PrimitiveArray<'a>> {
+        let to_subtract = T::try_from(to_subtract)?;
+
+        if to_subtract.is_zero() {
+            if let Some(min) = self.statistics().compute_as_cast(Stat::Min) {
+                let min: T = min;
+                if let (min, true) = min.overflowing_sub(&to_subtract) {
+                    vortex_bail!(
+                        "Integer subtraction over/underflow: {}, {}",
+                        min,
+                        to_subtract
+                    )
+                }
+            }
+            if let Some(max) = self.statistics().compute_as_cast(Stat::Max) {
+                let max: T = max;
+                if let (max, true) = max.overflowing_sub(&to_subtract) {
+                    vortex_bail!(
+                        "Integer subtraction over/underflow: {}, {}",
+                        max,
+                        to_subtract
+                    )
+                }
+            }
+            // if to_subtract is zero, skip operation
+            return Ok(self.clone());
+        }
+
+        let sub_vec: Vec<T> = if should_wrap {
+            let validity = self
+                .logical_validity()
+                .to_null_buffer()?
+                .expect("should_wrap only true if there are nulls");
```

Comment:
Why are we switching on a boolean if the the meaning of the boolean is equal to whether or not this null buffer exists? Can you just pass in Validity? Or LogicalValidity?